subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1620,14 +1620,28 @@ int ull_adv_init(void)
 	return 0;
 }
 
-int ull_adv_reset(void)
+uint8_t ll_adv_disable_all(void)
 {
 	uint8_t handle;
+	uint8_t err;
 
+	err = 0U;
 	for (handle = 0U; handle < BT_CTLR_ADV_SET; handle++) {
-		(void)disable(handle);
+		uint8_t ret;
+
+		ret = disable(handle);
+		if (ret) {
+			err = ret;
+		}
```

Comment:
If we have 3 sets and it disables set_0, but fails to disable set_1, should we not reject the command entirely and not disable any of the sets? 

The HCI spec doesn't say anything here, but since the host will not know which has failed or which was success, I think it is better to have it as all-or-none

Reply:
Currently `disable()` does not return error unless a debug `CONFIG_BT_CTLR_ADV_ENABLE_STRICT` is enabled.

By BT Spec. v5.4, Vol 4. Part E, 7.8.9 LE Set Advertising Enable command
> Disabling advertising when it is already disabled has no effect.

Reply:
> Currently disable() does not return error unless a debug CONFIG_BT_CTLR_ADV_ENABLE_STRICT is enabled.

That does not seem to be the case, as the function has 
```c
	if (adv->lll.conn) {
		/* Indicate to LLL that a cancellation is requested */
		adv->lll.conn->periph.cancelled = 1U;
		cpu_dmb();

		/* Check if a connection was initiated (connection
		 * establishment race between LLL and ULL).
		 */
		if (unlikely(adv->lll.conn->periph.initiated)) {
			return BT_HCI_ERR_CMD_DISALLOWED;
		}
	}
```

and 
```c
		ret = ull_ticker_status_take(ret, &ret_cb);
		if (ret) {
			mark = ull_disable_unmark(adv);
			LL_ASSERT(mark == adv);

			return BT_HCI_ERR_CMD_DISALLOWED;
		}
```

and several other error branches

Reply:
You are correct, these are corner cases where connection is established and/or advertising is terminated due to  max events or duration etc. I these cases, disable all adverising set should ignore them.

Reply:
Fair - But what about the case where `CONFIG_BT_CTLR_ADV_ENABLE_STRICT=y`? 

If the `disable` can only return an error if the set is already disabled or if it is being disabled by other means, then the disable all function should just completely ignore the return value, right?

Reply:
i think you are right, disable all should completely ignore the return value irrespective of `CONFIG_BT_CTLR_ADV_ENABLE_STRICT=y`; this Kconfig only makes sense under debug to validate individual set being disabled.